minio/credentials/providers.py

```diff
@@ -233,7 +241,7 @@ def retrieve(self):
             data["AccessKeyId"],
             data["SecretAccessKey"],
             session_token=data["Token"],
-        ), expiration + timedelta(minutes=5)
+        ), expiration - self._expiry_delta
```

Comment:
Would you please add a comment to explain why we are negating here ?

Reply:
We should bring the comment from minio-go

Reply:
This comment?
```
// SetExpiration sets the expiration IsExpired will check when called.
//
// If window is greater than 0 the expiration time will be reduced by the
// window value.
//
// Using a window is helpful to trigger credentials to expire sooner than
// the expiration time given to ensure no requests are made with expired
// tokens.
```

Reply:
Anything that would make sense to use this value

Reply:
@febg can you add the comment?